corehq/apps/app_manager/models.py

```diff
@@ -2087,6 +2088,14 @@ def wrap(cls, data):
             data['enum'] = sorted({'key': key, 'value': value}
                                   for key, value in data['enum'].items())
 
+        # Lazy migration: xpath expressions from format to first-class property
+        if data.get('format') == 'calculate':
+            property_xpath = PropertyXpathGenerator(None, None, None, super(DetailColumn, cls).wrap(data)).xpath
```

Comment:
Is it legit/allowed to used named params here? This is pretty unreadable.

Reply:
Agree this is pretty messy.

We could have this [__init__](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/detail_screen.py#L73) take keyword arguments instead of positional arguments.

Another option is to separate out https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/detail_screen.py#L563 into a helper function that takes ```column``` as an argument.

Reply:
Thoughts?

Reply:
I think either would be fine and would be more readable. Also this definitely does not need to block this PR.